packages/typespec-test/test/todo_non_branded/generated/typespec-ts/review/todo-non-branded.api.md

```diff
@@ -4,605 +4,239 @@
+    // (undocumented)
+    update: (id: number, patch: TodoItemPatch, options?: TodoItemsUpdateOptionalParams) => Promise<{
+        id: number;
+        title: string;
+        createdBy: number;
+        assignedTo?: number;
+        description?: string;
+        status: "NotStarted" | "InProgress" | "Completed";
+        createdAt: Date;
+        updatedAt: Date;
+        completedAt?: Date;
+        labels?: TodoLabels;
+    }>;
 }
 
-// @public (undocumented)
-export interface TodoItemsUpdateMediaTypesParam {
+// @public
+export interface TodoItemsUpdateOptionalParams extends OperationOptions {
     // (undocumented)
-    contentType: "application/merge-patch+json";
+    contentType?: string;
 }
 
-// @public (undocumented)
-export type TodoItemsUpdateParameters = TodoItemsUpdateMediaTypesParam & TodoItemsUpdateBodyParam & RequestParameters;
-
-// @public
-export type TodoLabel = string | string[] | TodoLabelRecord | Array<TodoLabelRecord>;
-
 // @public
-export type TodoLabelOutput = string | string[] | TodoLabelRecordOutput | Array<TodoLabelRecordOutput>;
-
-// @public (undocumented)
 export interface TodoLabelRecord {
     // (undocumented)
     color?: string;
     // (undocumented)
     name: string;
 }
 
-// @public (undocumented)
-export interface TodoLabelRecordOutput {
-    // (undocumented)
-    color?: string;
-    // (undocumented)
-    name: string;
-}
+// @public
+export type TodoLabels = string | string[] | TodoLabelRecord | TodoLabelRecord[];
 
-// @public (undocumented)
-export interface TodoPageOutput {
-    items: Array<TodoItemOutput>;
+// @public
+export interface TodoPage {
+    items: TodoItem[];
     // (undocumented)
     pagination: {
         pageSize: number;
         totalSize: number;
+        limit?: number;
+        offset?: number;
         prevLink?: string;
         nextLink?: string;
     };
 }
 
-// @public (undocumented)
+// @public
 export interface TodoUrlAttachment {
     description: string;
     url: string;
 }
 
-// @public (undocumented)
-export interface TodoUrlAttachmentOutput {
-    description: string;
-    url: string;
-}
-
-// @public (undocumented)
+// @public
 export interface User {
-    email: string;
-    password: string;
-    username: string;
-}
-
-// @public (undocumented)
-export interface UserCreatedResponseOutput {
     email: string;
     readonly id: number;
     password: string;
-    token: string;
     username: string;
+    validated: boolean;
 }
 
 // @public
-export interface UserExistsResponseOutput extends ErrorModelOutput {
-}
-
-// @public (undocumented)
-export interface UsersCreate {
-    // (undocumented)
-    post(options: UsersCreateParameters): StreamableMethod<UsersCreate200Response | UsersCreate409Response | UsersCreate422Response>;
-}
-
-// @public
-export interface UsersCreate200Response extends HttpResponse {
-    // (undocumented)
-    body: UserCreatedResponseOutput;
+export interface UserExistsResponse extends ApiError {
```

Comment:
We have a decision to not generate error models if it is not referenced - https://github.com/Azure/autorest.typescript/issues/2917.

For un-branded I would like to apply this rule - not generate these un-referenced models. Any concern?

Reply:
Renhe may need to confirm the priority with requirement first so we may could defer the conversation.